securesystemslib/interface.py

```diff
@@ -900,6 +900,260 @@ def import_ecdsa_privatekey_from_file(filepath, password=None):
+
+  return spx_key
+
+
+
+
+
+def import_spx_privatekey_from_file(filepath, password=None, prompt=False):
+  """
+  <Purpose>
+    Import the encrypted spx key file in 'filepath', decrypt it, and return
+    the key object in 'securesystemslib.formats.SPXKEY_SCHEMA' format.
+
+    The private key (may also contain the public part) is encrypted with AES
+    256 and CTR the mode of operation.  The password is strengthened with
+    PBKDF2-HMAC-SHA256.
+
+  <Arguments>
+    filepath:
+      <filepath> file, an RSA encrypted key file.
+
+    password:
+      The password, or passphrase, to import the private key (i.e., the
+      encrypted key file 'filepath' must be decrypted before the spx key
+      object can be returned.
+
+    prompt:
+      If True the user is prompted for a passphrase to decrypt 'filepath'.
+      Default is False.
+
+  <Exceptions>
+    securesystemslib.exceptions.FormatError, if the arguments are improperly
+    formatted or the imported key object contains an invalid key type (i.e.,
+    not 'spx').
+
+    securesystemslib.exceptions.CryptoError, if 'filepath' cannot be decrypted.
+
+  <Side Effects>
+    'password' is used to decrypt the 'filepath' key file.
+
+  <Returns>
+    An spx key object of the form:
+    'securesystemslib.formats.SPXKEY_SCHEMA'.
+  """
+
+  # Does 'filepath' have the correct format?
+  # Ensure the arguments have the appropriate number of objects and object
+  # types, and that all dict keys are properly named.
+  # Raise 'securesystemslib.exceptions.FormatError' if there is a mismatch.
+  securesystemslib.formats.PATH_SCHEMA.check_match(filepath)
+
+  if password and prompt:
+    raise ValueError("Passing 'password' and 'prompt' True is not allowed.")
+
+  # If 'password' was passed check format and that it is not empty.
+  if password is not None:
+    securesystemslib.formats.PASSWORD_SCHEMA.check_match(password)
+
+    # TODO: PASSWORD_SCHEMA should be securesystemslib.schema.AnyString(min=1)
+    if not len(password):
+      raise ValueError('Password must be 1 or more characters')
+
+  elif prompt:
+    # Password confirmation disabled here, which should ideally happen only
+    # when creating encrypted key files (i.e., improve usability).
+    # It is safe to specify the full path of 'filepath' in the prompt and not
+    # worry about leaking sensitive information about the key's location.
+    # However, care should be taken when including the full path in exceptions
+    # and log files.
+    # NOTE: A user who gets prompted for a password, can only signal that the
+    # key is not encrypted by entering no password in the prompt, as opposed
+    # to a programmer who can call the function with or without a 'password'.
+    # Hence, we treat an empty password here, as if no 'password' was passed.
+    password = get_password('Enter a password for an encrypted RSA'
+        ' file \'' + Fore.RED + filepath + Fore.RESET + '\': ',
```

Comment:
What's the reason you chose single quotes for these literals instead of using double quotes such as in `" file '" + ...` vs. `'file \''`?